### PR TITLE
Change buildsystem to latexmk

### PR DESCRIPTION
**Pull request by LeroyR**
Default Build sysyem changed to `latexmk` with `-Werror` on  `texlife/texlife2022` image, to catch missing refs like #851 during ci.
You can test the ci build locally with `latexdockercmd.sh`

- moved all "documents" into documents
  - let the makefile generate a rule for each tex file there
- removed makefiles
- add lint target to run chktex (as lint ci stage does)
- use stable ci image (texlive/texlive:TL2022-historic)
- update latexdockercmd.sh to just run "make all"

### CODE REVIEW

**Comment by LeroyR** [comment]
Ci fails until #851 is merged

**Comment by LeroyR** [comment]
Ci fails until #851 is merged

**Comment by MatthijsBurgh** [comment]
Am I correct, the CI fails because of the missing refs?

(Because there is also git safe directory messages.) (Preferably these are also gone.)

**Comment by LeroyR** [comment]
> Am I correct, the CI fails because of the missing refs?
> 
> (Because there is also git safe directory messages.) (Preferably these are also gone.)

Yes fails untils #851 is merged, as this uses `-Werror`

```
Latexmk: Summary of warnings from last run of *latex:
  Reference `award:hri' on page 10 undefined on input line 114
  Reference `rule:not_attending' on page 14 undefined on input line 32
  Reference `rule:start_signal' on page 14 undefined on input line 32
  Reference `rule:not_attending' on page 17 undefined on input line 14
  Reference `rule:start_signal' on page 17 undefined on input line 14
  Reference `rule:perceived_intelligence' on page 20 undefined on input line 29
  Reference `rule:not_attending' on page 20 undefined on input line 29
  Latex failed to resolve 20 reference(s)

 And 13 more --- see log file.
----------------------
This message may duplicate earlier message.
Latexmk: Failure in processing file 'documents/rulebook.tex':
   Warning(s) from latex (or c.) for 'documents/rulebook.tex'; treated as error
 ```
 
This `git safe directory` messages are from latex calling `git log something` to get date and revision number, but that is broken in CI (also broken on released master document (footer is `revision_\n0` instead of date) 

Github action/checkout is doing something (sparse checkout? and then mangles the repo (see Run actions/checkout@v4) 

```
Getting Git version info
Temporarily overriding HOME='/__w/_temp/749ac964-1caa-429b-8be4-0a0d0ccd22c1' before making global git config changes
Adding repository directory to the temporary git global config as a safe directory
/usr/bin/git config --global --add safe.directory /__w/RuleBook/RuleBook
Deleting the contents of '/__w/RuleBook/RuleBook'
```

**Review by johaq — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -45,6 +45,13 @@ jobs:
     - uses: actions/checkout@v4
       with:
         ref: gh-pages
+        
+    - name: Take ownership of the checkout directory (Git CVE-2022-24765)
+      run: chown --recursive --reference=/ .
+
+    - name: Allow for file ownership conflicts with Docker and GitHub Actions
+      run: git config --global --add safe.directory '*'
+
```
[the remark this answers is not in the document]
Comment (LeroyR) [reply .github/workflows/ci.yml:54]: copy pasted wrong

**Pull request merged**